app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -679,33 +690,48 @@ class NewMessageViewModel @Inject constructor(
         startWorkerCallback: () -> Unit,
     ) = globalCoroutineScope.launch(ioDispatcher) {
 
-        val draft = getLatestLocalDraft(draftLocalUuid) ?: return@launch
+        val localUuid = draftLocalUuid ?: return@launch
         val subject = subjectValue.ifBlank { null }?.take(SUBJECT_MAX_LENGTH)
 
-        draft.updateDraftFromLiveData(action, subject, uiBodyValue)
-
-        if (isFinishing && isSavingDraftWithoutChanges(draft, action, subject, uiBodyValue)) {
-            if (!arrivedFromExistingDraft) removeDraftFromRealm(draft.localUuid)
+        if (isFinishing && isSavingDraftWithoutChanges(action, subject, uiBodyValue)) {
```

Comment:
Why not try to skip the draft saving if there's no change anyway when placing the app in the background?

Reply:
If we do that with "Don't keep activities" activated, we're gonna loose the Draft.

Reply:
If the draft hasn't been modified (i.e. `isSavingDraftWithoutChanges`) there should not be anything to loose in all logic

Reply:
`isSavingDraftWithoutChanges ` doesn't mean there is nothing.
It only means that the Draft hasn't been modified by the user, but there is still a whole Draft that exists.

It will be lost, and we'll need to create it again when resuming the app.

Reply:
we agreed there shouldn't be a need to save it